generate_analysis_graphs.py

```diff
@@ -84,6 +85,84 @@
         individuals = TracedDataJsonIO.import_jsonl_to_traced_data_iterable(f)
     log.info(f"Loaded {len(individuals)} individuals")
 
+    log.info(f'Computing repeat and new participation per show ...')
+    # Computes the number of new and repeat consented individuals who participated in each radio show.
+    # Repeat participants are consented individuals who participated in previous shows prior to the target show.
+    # New participants are consented individuals who participated in target show but din't participate in previous shows.
+    repeat_new_participation_map = OrderedDict()  # of rqa_raw_field to participation metrics.
+
+    rqa_raw_fields =  [plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS]
+
+    for rqa_raw_field in rqa_raw_fields:
+        target_radio_show = rqa_raw_field  # radio show in which we are calculating repeat and new participation metrics for.
+
+        target_radio_show_participants = set()  # contains uids of individuals who participated in target radio show.
+        for ind in individuals:
+            if ind["consent_withdrawn"] == Codes.TRUE:
+                continue
+
+            if target_radio_show in ind:
```

Comment:
(I have a slight preference for checking the coded fields here, because then you don't have to worry about making the same judgements about missing keys, empty string messages etc. as you do in the pipeline. Instead, you would do it once in the pipeline then check to see if this field hasn't been coded as NA or NS. However, as this is a local solution, I'm not so fussed. Maybe at least add a TODO so that you know about this if you copy and paste to a future project. 
  In the long term, I think the solution is a centrally-defined responded() function (so then you can say if responded(ind, show)), but that is awaiting review from Luke and perhaps some further development/refinement).

Reply:
Totally agree. I have added a TODO for now.